### PR TITLE
ReactAsset improve error messages

### PR DESCRIPTION
**Pull request by cehan-Chloe**
Issue: https://github.com/player-ui/player/issues/376 

Test result: https://app.buildbuddy.io/invocation/8eb077c0-1586-4699-b732-c92c332e7697

What's changed: improve error msg thrown by ReactAsset. more info and react-like errors
Before: if no matched asset found, throw error: `No implementation found for id: bar-id type: bar`
After: 
1. add check for no assets in registry
image

2. modify existing error msg for no match asset found
image

### Change Type (required)
Indicate the type of change your pull request is:

- [ ] `patch`
- [x] `minor`
- [ ] `major`

### Does your PR have any documentation updates?
- [ ] Updated docs
- [x] No Update needed
- [ ] Unable to update docs

### CODE REVIEW

File: react/player/src/asset/index.tsx [hunk 1]
```diff
@@ -49,11 +49,30 @@ export const ReactAsset = (
...
+      See https://player-ui.github.io/latest/tools/cli#player-dependency-versions-check for tips about how to debug and fix this problem`);
+  }
+
   const Impl = registry?.get(unwrapped);
 
+  const matchList: object[] = [];
+
+  registry.forEach((asset) => {
+    matchList.push(asset.key);
+  });
```
Comment (KetanReddy) [inline react/player/src/asset/index.tsx:65]: Can we compute this only in the error case so we don't do this work if there is no error?

File: core/partial-match-registry/src/index.ts [hunk 2]
```diff
@@ -75,4 +75,9 @@ export class Registry<V> {
   clear() {
     this.store = createSortedArray<V>();
   }
+
+  /** Check if the registry is empty*/
+  isRegistryEmpty() {
```
Comment (KetanReddy) [inline core/partial-match-registry/src/index.ts:80]: @sugarmanz @hborawski Do we expose the partial match registry on every platform(It doesn't look like we do but I just wanted to check)? If so we should also add this to every platforms API.

File: react/player/src/asset/index.tsx [hunk 3]
```diff
@@ -49,11 +49,30 @@ export const ReactAsset = (
...
+
   if (!Impl) {
     throw Error(
-      `No implementation found for id: ${unwrapped.id} type: ${unwrapped.type}`,
+      `No implementation found for id: ${unwrapped.id} type: ${
+        unwrapped.type
+      }. This could happen for one of the following reasons: \n
+      1. You might not have the asset id: ${unwrapped.id} type: ${unwrapped.type} registered. \n
+      2. You might have multiple assets libraries in the same app. \n
+      See match list below for more information: \n
```
Comment (KetanReddy) [inline react/player/src/asset/index.tsx:74]:
```suggestion
      Registered Asset matching functions are listed below: \n
```

File: react/player/src/asset/index.tsx [hunk 4]
```diff
@@ -49,11 +49,30 @@ export const ReactAsset = (
...
+  registry.forEach((asset) => {
+    matchList.push(asset.key);
+  });
+
   if (!Impl) {
     throw Error(
-      `No implementation found for id: ${unwrapped.id} type: ${unwrapped.type}`,
+      `No implementation found for id: ${unwrapped.id} type: ${
+        unwrapped.type
+      }. This could happen for one of the following reasons: \n
```
Comment (adierkens) [inline react/player/src/asset/index.tsx:71]: Could we check for similarly named types and suggest alternatives? 

```
No implementation found for type: "txt" did you mean "text"?
```

File: core/partial-match-registry/src/index.ts (hunk 2 above)
Comment (hborawski) [reply core/partial-match-registry/src/index.ts:80]: https://github.com/player-ui/player/blob/bazel-6/ios/core/Sources/CorePlugins/PartialMatchFingerprintPlugin.swift#L75

We do have it in iOS, but it's not exposed to users + no Swift API currently

**Pull request closed without merging**